### DISCUSSION TITLE
Clp coinbrew install

### DISCUSSION THREAD

**Discussion by jpbrooks** [coin-or/Clp#174] [Q&A] [answered]
I tried to install with
wget https://raw.githubusercontent.com/coin-or/coinbrew/master/coinbrew
chmod u+x coinbrew
./coinbrew fetch Clp@master
./coinbrew build --prefix /home/jpbrooks Clp

and received the following error below.  Any help would be appreciated.

##################################################
##################################################

In file included from /usr/include/c++/4.8.2/algorithm:62:0,
                 from /home/jpbrooks/CoinUtils/src/CoinBronKerbosch.cpp:21:
/usr/include/c++/4.8.2/bits/stl_algo.h: In instantiation of ‘_RandomAccessIterator std::__unguarded_partition(_RandomAccessIterator, _RandomAccessIterator, const _Tp&, _Compare) [with _RandomAccessIterator = BKVertex*; _Tp = BKVertex; _Compare = bool (*)(BKVertex&, const BKVertex&)]’:
/usr/include/c++/4.8.2/bits/stl_algo.h:2296:78:   required from ‘_RandomAccessIterator std::__unguarded_partition_pivot(_RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = BKVertex*; _Compare = bool (*)(BKVertex&, const BKVertex&)]’
/usr/include/c++/4.8.2/bits/stl_algo.h:2337:62:   required from ‘void std::__introsort_loop(_RandomAccessIterator, _RandomAccessIterator, _Size, _Compare) [with _RandomAccessIterator = BKVertex*; _Size = long int; _Compare = bool (*)(BKVertex&, const BKVertex&)]’
/usr/include/c++/4.8.2/bits/stl_algo.h:5499:44:   required from ‘void std::sort(_RAIter, _RAIter, _Compare) [with _RAIter = BKVertex*; _Compare = bool (*)(BKVertex&, const BKVertex&)]’
/home/jpbrooks/CoinUtils/src/CoinBronKerbosch.cpp:362:70:   required from here
/usr/include/c++/4.8.2/bits/stl_algo.h:2266:34: error: invalid initialization of reference of type ‘BKVertex&’ from expression of type ‘const BKVertex’
    while (__comp(__pivot, *__last))
                                  ^
make[2]: *** [CoinBronKerbosch.lo] Error 1
make[1]: *** [all] Error 2
make: *** [all-recursive] Error 1

Build failed, see error output above

**Comment by svigerske** [comment, marked as answer]
Can you try if the following change in CoinUtils works?
```diff
--- a/src/CoinBronKerbosch.cpp
+++ b/src/CoinBronKerbosch.cpp
@@ -36,7 +36,7 @@ static void *xmalloc( const size_t size );
 static void *xcalloc( const size_t elements, const size_t size );
 static void *xrealloc( void *ptr, const size_t size );
 
-bool compareNodes(BKVertex &u, const BKVertex &v) {
+bool compareNodes(const BKVertex &u, const BKVertex &v) {
     return u.fitness >= v.fitness + BK_EPS;
 }
 
```

**Reply by jpbrooks** [reply]
That worked, thank you!
